clang/lib/Lex/Preprocessor.cpp

```diff
@@ -951,9 +951,16 @@ void Preprocessor::Lex(Token &Result) {
         break;
       [[fallthrough]];
     default:
-      TrackGMFState.handleMisc();
-      StdCXXImportSeqState.handleMisc();
-      ModuleDeclState.handleMisc();
+      if (tok::isPragmaAnnotation(Result.getKind())) {
```

Comment:
This is slightly hack. Let's add a new case for it and maybe we need to update the state machine if required.